core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -242,7 +246,8 @@ object DumpLogSegments {
                       nonConsecutivePairsForLogFilesMap: mutable.Map[String, List[(Long, Long)]],
                       isDeepIteration: Boolean,
                       maxMessageSize: Int,
-                      parser: MessageParser[_, _]): Unit = {
+                      parser: MessageParser[_, _],
+                      skipBatchMetadata: Boolean): Unit = {
```

Comment:
Should we skip `printBatchLevel` below if `skipBatchMetadata` is set?

Reply:
let me rename this to `--skip-record-metadata` to make it clearer. I think per-batch metadata is still useful (in particular, offset)